thumbfast.lua

```diff
@@ -888,7 +1018,318 @@ local function sync_changes(prop, val)
     dirty = true
 end
 
-local function file_load()
+local function get_thumb(atlas_path, atlas_idx, storyboard, thumbnail_size, storyboard_scale)
+    local atlas = io.open(atlas_path, "rb")
+    local atlas_filesize = atlas:seek("end")
+    local atlas_pictures = math.floor(atlas_filesize / (4 * thumbnail_size.w * thumbnail_size.h) + 0.5)
+    local stride = 4 * (thumbnail_size.w * math.min(storyboard.columns, atlas_pictures))
+    for pic = 0, atlas_pictures-1 do
+        local x_start = (pic % storyboard.columns) * thumbnail_size.w
+        local y_start = math.floor(pic / storyboard.columns) * thumbnail_size.h
```

Comment:
Because you want to move down by one thumbnail when you reach the end of the row (works fine most of the time as-is though, since `columns` and `rows` are usually equal for most videos)
```suggestion
        local y_start = math.floor(pic / storyboard.rows) * thumbnail_size.h
```

Reply:
There will probably be more correctness errors like this in a future commit, I had to change the `get_thumb` function a bunch to support rotated and flipped thumbnails. But that bug you pointed out still managed to survive through the rewrite, hahaha.

Reply:
The end of the row is reached when we exceed storyboard.columns. I'm pretty sure the current code is correct, and your suggestion is wrong. You'd even commented on 4b45873527b3dd8d6832dc27d0acd54476e5fcb9 letting me know I should use storyboard.columns.

Reply:
Yeah, not sure what I was thinking...